### ISSUE TITLE
Build fails when running unit tests

### ISSUE THREAD

**Issue by rookboom** [fable-compiler/fable-suave-scaffold#148]
### Description

When running build.cmd I get an error stating:
"d:\repos2\fable-suave-scaffold\test\ServerTests\Tests.fs(3,6): error FS0039: The namespace or module 'Expecto' is not defined."

From looking at the fsproj file, I cannot see where it is referenced.

### Repro steps

1. Step A
git clone 

2. Step B
run build.cmd from Command window.

### Expected behavior

I expected the build to pass without errors

### Actual behavior

There are test errors. Packages seems to be missing, although they are downloaded to packages\test\Expecto

### Known workarounds
None that I know of. I'm new to .Net core so I have no idea how to fix this.

### Related information 

* Operating system - Windows 10
* Branch - master
* .NET Runtime, CoreCLR or Mono Version
.NET Core 2.0
* Performance information, links to performance testing scripts

Here is a more detailed screen dump:
//================================================================
Starting Target: BuildServerTests (==> BuildClient)
dotnet build
Microsoft (R) Build Engine version 15.3.409.57025 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Server -> d:\repos2\fable-suave-scaffold\src\Server\bin\Debug\netcoreapp2.0\Server.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.10
Finished Target: BuildServerTests
Starting Target: RunServerTests (==> BuildServerTests)
dotnet run
d:\repos2\fable-suave-scaffold\test\ServerTests\Tests.fs(3,6): error FS0039: The namespace or module 'Expecto' is not defined. [d:\repos2\fable-suave-scaffold\test\ServerTests\ServerTests.fsproj]
d:\repos2\fable-suave-scaffold\test\ServerTests\Tests.fs(7,3): error FS0039: The value or constructor 'testList' is not defined. Maybe you want one of the following:   WishList [d:\repos2\fable-suave-scaffold\test\ServerTests\ServerTests.fsproj]
d:\repos2\fable-suave-scaffold\test\ServerTests\Program.fs(3,6): error FS0039: The namespace or module 'Expecto' is not defined. [d:\repos2\fable-suave-scaffold\test\ServerTests\ServerTests.fsproj]
d:\repos2\fable-suave-scaffold\test\ServerTests\Program.fs(7,9): error FS0039: The value or constructor 'runTestsWithArgs' is not defined. [d:\repos2\fable-suave-scaffold\test\ServerTests\ServerTests.fsproj]

The build failed. Please fix the build errors and run again.
Running build failed.
Error:
System.Exception: dotnet run failed
   at Microsoft.FSharp.Core.PrintfModule.PrintFormatToStringThenFail@1379.Invoke(String message)
   at FSI_0005.Build.runDotnet(String workingDir, String args)
   at FSI_0005.Build.clo@160-13.Invoke(Unit _arg12)
   at Fake.TargetHelper.runSingleTarget(TargetTemplate`1 target) in D:\code\fake\src\app\FakeLib\TargetHelper.fs:line 626

---------------------------------------------------------------------
Build Time Report
---------------------------------------------------------------------
Target               Duration
------               --------
Clean                00:00:00.0252372
InstallDotNetCore    00:00:00.0710740
InstallServer        00:00:07.0040820
InstallServerTests   00:00:11.5556242
InstallClientTests   00:00:05.2271147
InstallClient        00:00:07.6406079
BuildServer          00:00:11.6913353
BuildClient          00:00:21.4524359
BuildServerTests     00:00:07.5373696
RunServerTests       Failure
Total:               00:01:27.5267116
---------------------------------------------------------------------
Status:              Failure
---------------------------------------------------------------------
---------------------------------------------------------------------
  1) System.Exception: dotnet run failed
   at Microsoft.FSharp.Core.PrintfModule.PrintFormatToStringThenFail@1379.Invoke(String message)
   at FSI_0005.Build.runDotnet(String workingDir, String args)
   at FSI_0005.Build.clo@160-13.Invoke(Unit _arg12)
   at Fake.TargetHelper.runSingleTarget(TargetTemplate`1 target) in D:\code\fake\src\app\FakeLib\TargetHelper.fs:line 626
---------------------------------------------------------------------

**Comment by alfonsogarciacaro** [comment]
Nuget dependencies are managed by Paket, you can find them in the `paket.references` file.

This is strange, tests are running fine in AppVeyor. Is this happening with latest changes in the master branch of the repo?

**Comment by aprooks** [comment]
I can confirm this very same issue. Though it was working few days ago.

**Comment by jeremyabbott** [comment]
I can also confirm. I wasn't sure how to fix it though. I created a new .NET Core 2.0 console app with Paket and added expecto and everything worked as expected.

**Comment by alfonsogarciacaro** [comment]
I can repro on macOS. Maybe a Paket issue? (Pinging @matthid.) There is a [successful build in Appveyor](https://ci.appveyor.com/project/alfonsogarciacaro/fable-suave-scaffold/build/0.0.1.532) with Paket 5.89.1, but now Paket 5.91.0 seems not capable of restoring [the ServerTests](https://github.com/fable-compiler/fable-suave-scaffold/tree/master/test/ServerTests). After runing `dotnet restore` I cannot find any `obj/ServerTests.fsproj.references` file :confused:

**Comment by matthid** [comment]
Please see https://github.com/fsprojects/Paket/issues/2680 I think it might be the same problem (targets not updated + updated to netstandard20)

**Comment by alfonsogarciacaro** [comment]
Thanks for the pointers @matthid. I have deleted the `.targets` files and downloaded manually from Paket latest GH release. Now I get the following:

```
~/d/fable-suave-scaffold ❯❯❯ cd test/ServerTests/
~/d/f/t/ServerTests ❯❯❯ dotnet restore
  Paket version 5.91.0
  Performance:
   - Disk IO: 55 milliseconds
   - Runtime: 3 seconds
/Users/alfonsogarciacaronunez/dev/fable-suave-scaffold/.paket/Paket.Restore.targets(78,5): warning : Detected a change ('target-framework') in the project file '/Users/alfonsogarciacaronunez/dev/fable-suave-scaffold/src/Server/Server.fsproj', calling paket restore [/Users/alfonsogarciacaronunez/dev/fable-suave-scaffold/src/Server/Server.fsproj]
  Paket version 5.91.0
  Performance:
   - Disk IO: 243 milliseconds
   - Runtime: 4 seconds
/Users/alfonsogarciacaronunez/dev/fable-suave-scaffold/.paket/Paket.Restore.targets(82,5): error : A paket file for the framework 'netcoreapp2.0' is missing.Please delete 'paket-files/paket.restore.cached' and call 'paket restore'. [/Users/alfonsogarciacaronunez/dev/fable-suave-scaffold/src/Server/Server.fsproj]
```

I'm a good sport so I do what I'm told (delete `paket-files` and then run `mono .paket/paket.exe restore`, but paket.lock is not updated and I still get the same message when running `dotnet restore`.

**Comment by matthid** [comment]
I just noticed that the alpha release seems to have bundled the wrong targets file. that explains at least some of the problems we are seeing. Every update/install will currently write the "broken" targets file. Sorry for that.

Regarding your new error it seems like paket is not creating the required targets file in the obj directory. There should be a file named `Project.netcoreapp2.0.references` in the obj directory apparently there isn't?

Can you zip that `ServerTests` directory while I'm on it fixing bugs :)

**Comment by alfonsogarciacaro** [comment]
Here it is :)

[ServerTests.zip](https://github.com/fable-compiler/fable-suave-scaffold/files/1257730/ServerTests.zip)

**Comment by matthid** [comment]
Seems like I cannot reproduce I get (second time):
```
$ dotnet restore
C:\Program Files\dotnet\sdk\2.0.0\NuGet.targets(782,5): warning MSB3202: The project file "C:\proj\testing\src\Server\Server.fsproj" was not found. [C:\proj\testing\ServerTests\ServerTests\ServerTests.fsproj]
  Restore completed in 24.8 ms for C:\proj\testing\ServerTests\ServerTests\ServerTests.fsproj.
```

Maybe I need all relevant files or we wait for next update...

**Comment by forki** [comment]
Is this fixed?

Am 28.08.2017 8:21 nachm. schrieb "Matthias Dittrich" <
notifications@github.com>:

**Comment by aprooks** [comment]
I have pinpointed that this issue was introduced by paket 5.91.0 version. After rolling back to 5.90.1 build worked properly. @forki  should I raise an issue in paket repo?

**Comment by forki** [comment]
yes definetly. can you also please check latest beta?

**Comment by aprooks** [comment]
latest beta fails as well. Submitted bug

**Comment by matthid** [comment]
@alfonsogarciacaro If you say that fable related stuff is working with latest paket beta I'll try to get a stable out with latest changes (and then update https://github.com/fable-compiler/fable-suave-scaffold/pull/150)

**Comment by alfonsogarciacaro** [comment]
@matthid I'm not sure if it's just Fable related stuff, `dotnet build` doesn't work either as the references are missing, the ServerTests project in this repo is not a Fable project. @aprooks has tried with latest beta and it didn't work, I also tried in my work project this morning with 5.92.0-beta001 and failed (Same `Detected a change ('target-framework') in the project file` message) so I had to revert to 5.90.1 :confused: Looks more as it's related with netsandard projects.

**Comment by matthid** [comment]
@alfonsogarciacaro It was working for me with the PR (#150) like I said we need to update the framework restriction in the `Test` group.

**Comment by aprooks** [comment]
worked for me with framework restriction, but only after migrating projects to netcoreapp2.0

**Comment by matthid** [comment]
Yes we probably want to remove the restriction completely.

**Comment by aprooks** [comment]
yes compiles without restriction as well. 

Another issue which is relevant (I guess) `dotnet watch run` fails with 

```
> C:\Users\admin\AppData\Local\dotnetcore\dotnet.exe --version                                                                                                 
2.0.0                                                                                                                                                          
[master ≡ +0 ~4 -0 !] D:\sources\oss\fable-scaff\src\Server                                                                                                    
> C:\Users\admin\AppData\Local\dotnetcore\dotnet.exe watch run                                                                                                 
watch : Error(s) finding watch items project file 'Server.fsproj'                                                                                              
watch : MSBuild output from target 'GenerateWatchList':                                                                                                        
watch :                                                                                                                                                        
watch :    Build started 29-Aug-17 23:45:03.                                                                                                                   
watch :         1>Project "D:\sources\oss\fable-scaff\src\Server\Server.fsproj" on node 1 (GenerateWatchList target(s)).                                       
watch :         1>D:\sources\oss\fable-scaff\src\Server\Server.fsproj : error MSB4057: The target "GenerateWatchList" does not exist in the project.           
watch :         1>Done Building Project "D:\sources\oss\fable-scaff\src\Server\Server.fsproj" (GenerateWatchList target(s)) -- FAILED.                         
watch :                                                                                                                                                        
watch :    Build FAILED.                                                                                                                                       
watch :                                                                                                                                                        
watch :           "D:\sources\oss\fable-scaff\src\Server\Server.fsproj" (GenerateWatchList target) (1) ->                                                      
watch :             D:\sources\oss\fable-scaff\src\Server\Server.fsproj : error MSB4057: The target "GenerateWatchList" does not exist in the project.         
watch :                                                                                                                                                        
watch :        0 Warning(s)                                                                                                                                    
watch :        1 Error(s)                                                                                                                                      
watch :                                                                                                                                                        
watch :    Time Elapsed 00:00:00.26                                                                                                                            
watch :                                                                                                                                                        
watch : Fix the error to continue or press Ctrl+C to exit.                                                                                                     
                                                                                                                                                               
```

**Comment by matthid** [comment]
@alfonsogarciacaro There is still the broken watch mode here. Also it seems like the broken framework restriction has been there since 6 months! Any idea what we can do?

**Comment by alfonsogarciacaro** [comment]
@matthid I see, thanks! I can confirm that it's working after merging your PR 😄 

The `Run` build target (which calls `dotnet watch msbuild /t:TestAndRun`) also works for me.

**Comment by matthid** [comment]
@alfonsogarciacaro Yes but a lot of people who used this template are broken with next paket update (or automatically with magic mode) and don't know what to do, right?

**Comment by alfonsogarciacaro** [comment]
@matthid But that would be solved with the `version 5.92.0-beta001` statement or when the next stable Paket version is released, right? I haven't done anything but merging the PR, pulling and running `./build.sh`. 

About the framework restriction, the ServerTests were recently updated to netcoreapp2 (it was not possible to do the same with UITests because of canopy I think), so I guess that's the reason it was not a problem before.

**Comment by matthid** [comment]
Yes but ServerTests was netcoreapp1.0 before that, correct? Or was it a full framework project?

Also I'm not talking about new users but existing ones who have used this project in the past. Do you think they merge in updates from here? Is that the recommended way?

**Comment by matthid** [comment]
Tbh I'm not even sure what our options are, ignoring framework restrictions for "netcore" apps is not really a sane option from paket's point of view.

**Comment by alfonsogarciacaro** [comment]
ServerTests was a full framework project before indeed. I was just trying to update everything to netcore after dotnet SDK 2 was released, but had issues with the UITests project.

About existing users, I hope they are pulling changes from this repo from time to time, we don't have another way to propagate fixes :/ In any case, the problem should only have happened during the time window when ServerTests was updated to netcore2 and Paket 5.91.0 was released, so hopefully it didn't affect many users 🙏

**Comment by matthid** [comment]
Ok I think it is fine then. I didn't actually test but I think 5.91 should be working as well with the correct restrictions. Thanks for clarifying.

**Comment by cartermp** [comment]
Verified that this is also fixed for vendors.
